components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/validators/TokenValidationHandler.java

```diff
@@ -559,6 +559,10 @@ private OAuth2IntrospectionResponseDTO validateAccessToken(OAuth2TokenValidation
         }
 
         addAllowedScopes(messageContext, requestedAllowedScopes.toArray(new String[0]));
+        if (accessTokenDO != null) {
+            addAllowedScopes(introResp, accessTokenDO, requestedAllowedScopes.toArray(new String[0]));
```

Comment:
shall we add a comment what is the difference between these two addAllowedScopes methods?

Reply:
Updated